rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/handlers/ResetUIHandler.xtend

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.handlers
+
+import java.util.function.Function
+import javax.inject.Inject
+import org.eclipse.e4.core.di.annotations.Execute
+import org.eclipse.e4.ui.model.application.MApplication
+import org.eclipse.e4.ui.model.application.ui.MElementContainer
+import org.eclipse.e4.ui.model.application.ui.MUIElement
+import org.eclipse.e4.ui.model.application.ui.advanced.MPerspective
+import org.eclipse.e4.ui.model.application.ui.advanced.MPerspectiveStack
+import org.eclipse.e4.ui.model.application.ui.basic.MPartStack
+import org.eclipse.e4.ui.model.application.ui.basic.MWindow
+import org.eclipse.e4.ui.workbench.IPresentationEngine
+import org.eclipse.e4.ui.workbench.modeling.EModelService
+import org.eclipse.e4.ui.workbench.modeling.EPartService
+import org.eclipse.swt.widgets.Display
+import org.eclipse.swt.widgets.Shell
+import org.slf4j.LoggerFactory
+import org.testeditor.rcp4.Constants
+
+class ResetUIHandler {
+
+	static val logger = LoggerFactory.getLogger(ResetUIHandler)
+	@Inject MApplication application
+
+	@Execute
+	public def void resetUI(EModelService modelService, EPartService partService) {
+		val window = application.getChildren().get(0) as MWindow
+		val perspectiveStack = modelService.find(Constants.MAIN_PERSPECTIVE_STACK_ID, application) as MPerspectiveStack
+		val originalPerspective = modelService.cloneSnippet(application, Constants.PERSPECTIVE_ID,
+			window) as MPerspective
+		if (originalPerspective === null) {
+			logger.warn('Could not reset UI since no original perspective has been saved (see SaveUIHandler.saveUI()).')
+			return
+		}
+
+		if (perspectiveStack.selectedElement !== null) {
```

Comment:
!== is the xtend variant of not identical, === is the equivalent of identical where == and != is intepreted as a call to compare (which does not make sense here)